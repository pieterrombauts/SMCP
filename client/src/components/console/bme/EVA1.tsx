import React from 'react';
import { bmeEVASuitDisplay } from '../customTypes'

interface EVA1Props {
  values: bmeEVASuitDisplay;
};

const EVA1: React.FC<EVA1Props> = (props) => {
  return (
    <g id="bme-EVA-suit-EVA1">
      <text x="13%" y="12%" fill="white" fontSize="50" fontFamily="Verdana">EVA 1</text>

      <text x="38%" y="26%" fill="white" fontSize="25" fontFamily="Verdana">{props.values.eva1.helmetPressure}</text>
      <text x="38%" y="41%" fill="white" fontSize="25" fontFamily="Verdana">{props.values.eva1.o2}</text>
      <text x="38%" y="58%" fill="white" fontSize="25" fontFamily="Verdana">{props.values.eva1.water}</text>
      <text x="38%" y="72%" fill="white" fontSize="25" fontFamily="Verdana">{props.values.eva1.batteryCharge}</text>
      <text x="38%" y="86%" fill="white" fontSize="25" fontFamily="Verdana">{props.values.eva1.pressure}</text>
      <text x="36%" y="97%" fill="#7ef56e" fontSize="25" fontFamily="Verdana"> ACTIVE</text>

      <image x="-13%" y="18%" width={800} height={800} href='/media/Bme_EVA_Suit_1.png'></image>

    </g>
  )
};

export default EVA1;