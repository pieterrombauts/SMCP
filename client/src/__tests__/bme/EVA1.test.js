import React from "react";
import 'jest-styled-components'
import toJson from 'enzyme-to-json'
import EVA1 from '../../components/console/bme/EVA1'
import { shallow, mount, render} from 'enzyme';

const defaultValues = {
  eva1: {
      helmetPressure: 0,
      o2: 65,
      water: 80,
      batteryCharge: 67
  },

  eva2: {
      helmetPressure: 0,
      o2: 65,
      water: 80,
      batteryCharge: 68
  }
}

describe("EVA1 display", () => {
  it('renders correctly', () => {
    const wrapper = shallow(<EVA1 values={defaultValues} />)
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  })
})