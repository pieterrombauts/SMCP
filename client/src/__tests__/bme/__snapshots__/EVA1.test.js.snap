// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EVA1 display renders correctly 1`] = `
<g
  id="bme-EVA-suit-EVA1"
>
  <text
    fill="white"
    fontFamily="Verdana"
    fontSize="50"
    x="4%"
    y="12%"
  >
    EVA 1 - Luca
  </text>
  <text
    fill="white"
    fontFamily="Verdana"
    fontSize="25"
    x="38%"
    y="26%"
  >
    0
  </text>
  <text
    fill="white"
    fontFamily="Verdana"
    fontSize="25"
    x="38%"
    y="41%"
  >
    65
  </text>
  <text
    fill="white"
    fontFamily="Verdana"
    fontSize="25"
    x="38%"
    y="58%"
  >
    80
  </text>
  <text
    fill="white"
    fontFamily="Verdana"
    fontSize="25"
    x="38%"
    y="72%"
  >
    67
  </text>
  <text
    fill="#7ef56e"
    fontFamily="Verdana"
    fontSize="25"
    x="36%"
    y="86%"
  >
     ACTIVE
  </text>
  <image
    height={800}
    href="/media/Bme_EVA_Suit_1.png"
    width={800}
    x="-13%"
    y="18%"
  />
</g>
`;
